### ISSUE TITLE
Update Alert `role`

### ISSUE THREAD

**Issue by huyenltnguyen** [freeCodeCamp/ui#132]
## Description

The `Alert` component always has the `alert` role:

https://github.com/freeCodeCamp/ui/blob/46150e76c8eaea15570f26001463d192f586b67a/src/alert/alert.tsx#L36

However, the `alert` role is quite disruptive to non-sighted users. The role is equivalent to setting `aria-live="assertive"`, which interrupts the user's current task and announces the message immediately. For this reason, the role should only be used for time-sensitive messages, typically error messages.

I'd propose that we change the component's `role` as follows:
- `role` is `alert` if the variant is `danger`
- `role` is `status` if the variant is `info`, `success`, or `warning`

---
Additionally, we should consider creating a separate component for callout boxes:

<details>
  <summary>Screenshots</summary>

| Variant | Screenshot |
| --- | --- |
| Info | <img width="789" alt="Screenshot 2024-05-15 at 14 23 32" src="https://github.com/freeCodeCamp/ui/assets/25715018/cedc0680-17d0-4d54-8470-9786ca282932"> |
| Danger | <img width="788" alt="Screenshot 2024-05-15 at 14 26 03" src="https://github.com/freeCodeCamp/ui/assets/25715018/79083327-38fd-488f-bfe1-01a77f25c401"> |
</details>

We're using `Alert` for those messages (to achieve consistent styles), but I think the content of those boxes should be read as part of the page content, rather than as a notification / alert.

The new callout component can have the same set of variants that Alert has, but renders its content in a plain `div`, without any roles.

## References

- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/status_role

**Comment by ahmaxed** [comment]
Would it be possible to just adjust the role for info, success, or warning instead of creating a new component?
